File: framework/languages/java/e2e/SpotService/Client/src/main/java/systems/zlink/e2e/spotservice/client/Scenarios/SmC1Scenario.java
package systems.zlink.e2e.spotservice.client.Scenarios;

import systems.zlink.e2e.spotservice.shared.Contracts;

public final class SmC1Scenario extends SpotServiceScenarioContext {
    private SmC1Scenario(SpotServiceScenarioContext context) {
        super(context);
    }

    public static void run(SpotServiceScenarioContext context) {
        new SmC1Scenario(context).execute();
    }

    private void execute() {
        Contracts.StateRes after = eventually(() -> requestState("room-a", "after-timeout", REQUEST_TIMEOUT));
        ensure(after.value().contains("after-timeout"), "SM-C1 post-timeout request failed");
        System.out.println("scenario SM-C1 passed");
        System.out.println("scenario SM-C1-normal passed");

    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Client/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/client/scenarios/SmC1Scenario.kt
package systems.zlink.e2e.kotlin.spotservice.client.scenarios

import systems.zlink.e2e.kotlin.spotservice.client.support.ensure
import systems.zlink.e2e.kotlin.spotservice.client.support.eventually
import systems.zlink.e2e.kotlin.spotservice.client.support.expectFailure
import systems.zlink.e2e.kotlin.spotservice.client.support.SpotHttpDriver

internal object SmC1Scenario {
    suspend fun runSend(spots: SpotHttpDriver) {
        spots.sendState("room-a", "cmd-c1")
        println("scenario SM-C1-send passed")
    }

    suspend fun runTimeout(spots: SpotHttpDriver) {
        expectFailure {
            spots.requestSlow("room-a", "late", 100)
        }
        println("scenario SM-C1-timeout passed")
    }

    suspend fun runNormal(spots: SpotHttpDriver) {
        val after = eventually {
            spots.requestState("room-a", "after-timeout")
        }
        ensure(after.value.contains("after-timeout"), "SM-C1 post-timeout request failed")
        println("scenario SM-C1 passed")
        println("scenario SM-C1-normal passed")
    }
}
